File: NowCoder/2020RealTime/vipkid/1.js
function outputSum1(num) {
    var res = [];

    var a=[];
    for (let i = 0; i < 10000; i++) {
        a[i]=0;
        
    }
    var y;
    var ss = 0;
    function dfs(n, cnt, pre) {
        if (n == 0) {
            ss++;
            var temp="";
            for (var i = 1; i < cnt; i++) {
                temp+=a[i]+"+";
            }
            temp+=a[cnt];
            if (ss % 4 == 0)
                res.push(temp);
            else {
                if (a[1] != y)
                    temp="";
            }
        }
        for (var i = pre; i <= n; i++) {
            a[++cnt] = i;
            dfs(n-i, cnt, i);
            cnt--;
        }
    }
    dfs(num,0,1);
    return res;
}


function outputSum(num) {
    var res = [];
    var back=[];
    var y=0,count=0,pos=-1;
    var n=num;
    function dfs(x) {
        if(y===n){
            count++;
            temp="";
            for (var i = 0; i < pos; i++) {
                temp+= res[i]+"+";
            }
            temp+=res[pos];
            
            if(back.indexOf(temp)===-1){
                back.push(temp);
            }
            
            return;
        }
        if(y>n){
            return;
        }
        for (let i =x; i < n+1; i++) {
            res[++pos]=i;
            y+=i;
            dfs(i);
            pos--;
            y-=i;
            
        }

    }
    dfs(1);
    return back;
}

// var num = readInt();
// var res = outputSum(num);
// for (var i = 0; i < res.length; i++) {
//     print(res[i]);
// }


var num=4;
var res = outputSum2(num);
for (var i = 0; i < res.length; i++) {
    console.log(res[i]);
}

File: NowCoder/2020RealTime/vipkid/1.c
#include <iostream>
#include<bits/stdc++.h>
using namespace std;
int a[1999999]={0};
int y;
int ss =0 ;
void dfs(int n,int cnt,int pre)
{
if(n==0)
{
ss++;
printf("%d=",y);
for(int i=1;i<cnt;i++)
{
printf("%d+",a[i]);
}
printf("%d",a[cnt]);
if(ss%4==0)
printf("\n");
else
{
if(a[1]!=y)
printf(";");
}
}
for(int i=pre;i<=n;i++)
{
a[++cnt] = i;
dfs(n-i,cnt,i);
cnt--;
}
}
int main()
{
int cnt =0 ;
int n;
cin>>n;
y = n;
dfs(n,0,1);
return 0;
}